packages/turf-line-chunk/index.ts

```
@@ -57,6 +57,12 @@ function lineChunk<T extends LineString | MultiLineString>(
 
   // Flatten each feature to simple LineString
   flattenEach(geojson, (feature: Feature<T>) => {
+    if (feature.geometry.type !== "LineString") {
+      throw new Error(
+        "Only LineString and MultiLineString geometry types are supported"
+      );
+    }
```

Open to handling points differently here, but this has the nice property of aligning the TypeScript strictness with the runtime behavior.

---

In general, what do we want Turf's runtime behaviour to be when it's gets unexpected / inappropriate input? Halt on unexpected input? Or soldier on silently (or with a warning)?

---

> points and polygons should be considered invalid and strongly rejected. But we should take a more moderate stance on the container shape

This seems a good balance. Caller has to be more alert about what they pass in. So for a geometry collection containing 99 lines and 1 point, we would throw?

> coming back with yet another PR when we do a major break

As is I think this would have to be a major release candidate:
- now throws where we previously silently ignored invalid geometries
- narrower input types could break builds

To keep moving forward between majors then, do we make this PR about the helpers.geometryCollection part? That doesn't narrow input types, and doesn't broaden the return type either. Runtime behaviour would stay the same.

---

Got eager and broke that out here https://github.com/Turfjs/turf/pull/2971